src/CompilingMatcher.php

```diff
@@ -51,11 +57,17 @@ class CompilingMatcher
      */
     public static function match(ConstraintInterface $constraint, $operator, $version)
     {
+        $resultCacheKey = $operator.$constraint.$version;
```

Comment:
I think you need a delimiter between the constraint and the version. Otherwise, `>=2.11` and `1.0` will share the cache key with `>=2.1` and `11.0` (it might require doing that in a different segment of the version used by the constraint depending on whether normalization is applied, but that's still possible)

Reply:
Excellent find! Fixed in https://github.com/composer/semver/pull/132